src/http_server.cc

```diff
@@ -2765,6 +2765,38 @@ HTTPAPIServer::EVBufferToRawInput(
   return nullptr;  // success
 }
 
+struct HeaderSearchPayload {
+  const char* regex_;
+  TRITONSERVER_InferenceRequest* request_;
+  TRITONSERVER_Error* error_;
+};
+
+int
+ForEachHeader(evhtp_header_t* header, void* arg)
+{
+  HeaderSearchPayload* header_search_payload =
+      reinterpret_cast<HeaderSearchPayload*>(arg);
+
+  TRITONSERVER_InferenceRequest* irequest = header_search_payload->request_;
+  const char* pattern = header_search_payload->regex_;
+
+  std::regex base_regex(pattern);
```

Comment:
Similar to other latency/perf comment, I think this line is expensive (compiling regex). If the regex pattern is the same everytime on server startup (since CLI arg is passed at startup), I think it may be significantly more performant to compile the regex once upfront, and then re-use it.

Reply:
Changed to prefix string matching 👍